lang/py/avro/test/test_io.py

```diff
@@ -80,7 +80,7 @@
         '{"type": "long", "logicalType": "timestamp-micros"}',
         datetime.datetime(2000, 1, 18, 2, 2, 1, 123499, tzinfo=avro.timezones.tst)
     ),
-    ('{"type": "string", "logicalType": "uuid"}', u'12345abcd'),
+    ('{"type": "string", "logicalType": "uuid"}', u'570feebe-2bbc-4937-98df-285944e1dbbd'),
```

Comment:
For the test case, should this be:
```suggestion
    ('{"type": "string", "logicalType": "uuid"}', uuid.UUID('570feebe-2bbc-4937-98df-285944e1dbbd')),
```
I'm not _sure_ if this suggestion is correct, but it's what I would expect once UUID logical type is implemented!

Since this is a new feature, should we go with the expectation that the in-memory representation of a UUID datum is a UUID object instead of a string?  Should we accept either/or when serializing, but always deserialize to one or the other?  What do you think?